common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -190,6 +217,52 @@ public void onFailure(Throwable t) {
     }
   }
 
+  private <T> void retry(
+      final int retryCountValue,
+      final int saslRetryCountValue,
+      final int maxRetries,
+      final boolean enableSaslRetries,
+      int delayMs,
+      Supplier<CompletableFuture<T>> action,
+      CompletableFuture<T> future) {
+    action.get()
+            .thenAccept(future::complete)
+            .exceptionally(e -> {
+              int retryCount = retryCountValue;
+              int saslRetryCount = saslRetryCountValue;
+              boolean isIOException = e instanceof IOException
+                      || (e.getCause() != null && e.getCause() instanceof IOException);
```

Comment:
`null` check is not required.

```suggestion
              boolean isIOException = e instanceof IOException || e.getCause() instanceof IOException;
```